andreeapetcu/assignment2/Ancestors.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * ASSUMPTIONS : the binary tree does not contain duplicate values
+ */
+
+import java.util.ArrayList;
+
+public class Ancestors<T> {
+
+  /**
+   * Wraper class that contains the ancestors of a node and a flag for the existence of a node
+   * with the given key.
+   */
+  class Result {
```

Comment:
FYI: It's normally good practice to make inner classes static if you can.
This is explained in more detail in "Effective Java 2nd edition" item 22. It helps with various memroy things.